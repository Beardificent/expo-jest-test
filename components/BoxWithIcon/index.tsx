import { Box, HStack, Icon, Text } from "native-base";
import { FC } from "react";
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5'

interface BoxWithIconProps {
  iconName: string;
  text: string;
}

const BoxWithIcon: FC<BoxWithIconProps> = ({ iconName, text }) => {
  return (
    <Box w={"fit-content"} borderWidth={1} p={2} data-testid="box-icon">
      <HStack alignItems={'center'}>
        <Icon
          color={"coolGray.700"}
          size={"2xl"}
          as={FontAwesome5}
          name={iconName}
        />
        <Text children={text} />
      </HStack>
    </Box>
  );
};

export default BoxWithIcon;
