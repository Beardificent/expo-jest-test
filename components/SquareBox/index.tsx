import { FC } from "react";
import { Box, View } from "native-base";
import React from "react";
import { INativebaseConfig } from "native-base"; // Update the import path for INativebaseConfig
import { NativeBaseProvider } from "native-base"; // Update the import path for NativeBaseConfigProvider

interface SquareBoxProps {
  children: any;
}
const SquareBox: FC<SquareBoxProps> = ({ children }) => {

  return (
      <Box w={'fit-content'} h={100} flexDir={'row'} backgroundColor={'blue.300'} borderWidth={1} p={2}>
        {children}
      </Box>
  );
};

export default SquareBox;
