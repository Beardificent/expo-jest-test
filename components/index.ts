import SquareBox from "./SquareBox";
import BoxWithIcon from "./BoxWithIcon";

export {SquareBox, BoxWithIcon}