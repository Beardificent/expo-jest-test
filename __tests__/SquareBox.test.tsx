import renderer from "react-test-renderer";
import { SquareBox } from "../components";
import { Link, NativeBaseProvider } from "native-base"; //mocking component would lead to no longer needing the NativeBaseProvider

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

describe("Squarebox", () => {
  it("renders SquareBox with NativeBaseProvider", () => {
    const expectedComponent = renderer.create(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <SquareBox
          children={<Link href="http://www.facebook.com">Facebook</Link>}
        />
      </NativeBaseProvider>
    );
    expect(expectedComponent).toBeTruthy();
    expect(expectedComponent).toMatchSnapshot();
  });

  it("renders SquareBox with NativeBaseProvider + correct property values", () => {
    const expectedComponent = renderer.create(
      <NativeBaseProvider initialWindowMetrics={inset}>
        <SquareBox children={<Link href="">Foobar</Link>} />
      </NativeBaseProvider>
    );
    expect(
      expectedComponent.root.findByType(SquareBox).props.children.type
    ).toBe(Link);
    expect(expectedComponent.root.findByType(Link).props.children).not.toBe(
      "Facebook"
    );
    expect(expectedComponent.root.findByType(Link).props.children).toBe(
      "Foobar"
    );
    expect(expectedComponent).toMatchSnapshot();
  });
});
