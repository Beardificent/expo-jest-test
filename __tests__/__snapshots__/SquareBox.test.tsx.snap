// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Squarebox renders SquareBox with NativeBaseProvider + correct property values 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "backgroundColor": "#93c5fd",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "fit-content",
      }
    }
  >
    <View
      accessibilityRole="link"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      dataSet={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "height": "auto",
          "width": "auto",
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": "underline",
          }
        }
      >
        Foobar
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;

exports[`Squarebox renders SquareBox with NativeBaseProvider 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "backgroundColor": "#93c5fd",
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 100,
        "paddingBottom": 8,
        "paddingLeft": 8,
        "paddingRight": 8,
        "paddingTop": 8,
        "width": "fit-content",
      }
    }
  >
    <View
      accessibilityRole="link"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      dataSet={{}}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "height": "auto",
          "width": "auto",
        }
      }
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#171717",
            "fontFamily": undefined,
            "fontSize": 14,
            "fontStyle": "normal",
            "fontWeight": "400",
            "letterSpacing": 0,
            "lineHeight": 21,
            "textDecorationLine": "underline",
          }
        }
      >
        Facebook
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
